fsw/src/nav_app.c

```diff
@@ -259,7 +252,12 @@ static void NAV_ReadAvailableData(uint8 DataBuffer[], int32 *DataLen)
 
     /* check how many bytes are waiting on the uart */
     *DataLen = uart_bytes_available(NAV_UART.handle);
-    //OS_printf("NAV_ReadAvailableData(): gps messages waiting: %ld bytes\n", (long int)*DataLen);
+    //OS_printf("NAV_ReadAvailableData(): gps bytes waiting: %d \n", DataLen);
+
+    if (*DataLen > NAV_BUFFER_SIZE)
+    {
+        *DataLen = NAV_BUFFER_SIZE;
+    }
```

Comment:
This whole merge was actually necessary due to an increase in the maximum UART buffer size in NOS3 overflowing the allocated buffer.